core/src/main/scala/shapeless/syntax/hlists.scala

```diff
@@ -738,4 +738,9 @@ final class HListOps[L <: HList](l : L) extends Serializable {
    * Returns all combinations of exactly length `N` of elements from this `Hlist`
    */
   def combinations(n: Nat)(implicit combinations: Combinations[n.N, L]): combinations.Out = combinations(l)
+
+  /**
+   * Converts this `HList` into a nested pair
+   */
+  def nestedProduct(implicit hListToProduct: HListToProduct[L]): hListToProduct.Out = hListToProduct(l)
```

Comment:
```suggestion
  def toNestedProduct(implicit hListToProduct: HListToProduct[L]): hListToProduct.Out = hListToProduct(l)
```

I would also rename then the corresponding one to `toNestedEither`.